### PR TITLE
Add mergify.io automation

### PR DESCRIPTION
**Pull request by mwachs5**
This PR adds mergify (https://mergify.io/) rules to enable automated merge-on-pass and smart ordering of PR merges. It would also enable backporting to release branches if that were a thing.

~I am marking this as DRAFT because I will be playing around with this on a personal fork of this repo.~

**Related issue**:  None

**Type of change**: other enhancement

**Impact**: no functional change

**Development Phase**: proposal

**Release Notes**

mergify.io automation added to enable simpler pull request merging and potentially backporting to stable branches.

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+pull_request_rules:
+  - name: automatic squash-and-merge on CI success and review
+    conditions:
+      - status-success=continuous-integration/travis-ci/pr
+      - "#approved-reviews-by>=1"
+      - "#changes-requested-reviews-by=0"
+      - base=master
+      - label="Please Merge"
+      - label!="DO NOT MERGE"
```
Comment (mwachs5) [inline .mergify.yml:16]:
```suggestion
      - label="MERGE WITH MERGIFY"
      - label!="DONT MERGE"
```

File: .mergify.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      - status-success=continuous-integration/travis-ci/pr
+      - "#approved-reviews-by>=1"
+      - "#changes-requested-reviews-by=0"
+      - base=master
+      - label="MERGE WITH MERGIFY"
+      - label!="DONT MERGE"
+      - label!="bp-conflict"
+    actions:
+      merge:
+        method: squash
```
Comment (mwachs5) [inline .mergify.yml:20]:
```suggestion
        method: merge
```

File: .mergify.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
+pull_request_rules:
+  - name: remove outdated reviews
+    conditions:
+      - base=master
+    actions:
+      dismiss_reviews:
+
+pull_request_rules:
+  - name: automatic squash-and-merge on CI success and review
```
Comment (mwachs5) [inline .mergify.yml:9]:
```suggestion
  - name: automatic merge on CI success and review
```

**Comment by mwachs5** [comment]
One thing I am not sure is the rule about the CI checks. Currently it's Travis;

`status-success=continuous-integration/travis-ci/pr`

How do I set it to go with our current GitHub actions CI...

**Comment by sequencer** [comment]
From the last dev-meeting, we decide to not using mergify currently, maybe after refactors, RC can be split to small repositories, then we can have a version scheme at that time.

**Pull request closed without merging**